Support PR trigger of ARM JIT stress jobs

BruceForstall opened the pull request:
Add a list of jobs known to the ARM64 CI, and allow those to be triggered.

Also, (1) allow Release triggered jobs to run tests (before, the label said
"Build and Test", but no test was done), (2) let arm64 debug do a PRI 1 test
build, like ARM32 (reduce differences between these similar tasks).

BruceForstall commented:
@dotnet-bot test CI please

BruceForstall commented:
@jashook @RussKeldorph PTAL

BruceForstall commented:
@dotnet-bot test CI please

jashook commented on netci.groovy:
```diff
@@ -1133,30 +1178,28 @@ def static addTriggers(def job, def branch, def isPR, def architecture, def os,
-                            // Stress jobs will will run on private trigger and run tests.
-                            Utilities.addPrivateGithubPRTriggerForBranch(job, branch, contextString,
-                            "(?i).*test\\W+${os}\\W+${architecture}\\W+${configuration}\\W+${scenario}.*", null, arm64Users)
+                        default:
```
Nit: add a comment mentioning stress jobs will use this code path.

jashook commented on netci.groovy:
```diff
@@ -1242,11 +1285,11 @@ def static addTriggers(def job, def branch, def isPR, def architecture, def os,
-                        case 'gcstress0xc':
-                            Utilities.addPrivateGithubPRTriggerForBranch(job, branch, contextString,
-                            "(?i).*test\\W+${os}\\W+${architecture}\\W+${configuration}\\W+${scenario}.*", null, arm64Users)
+                        default:
```
Same nit as the arm32 change

jashook commented on netci.groovy:
```diff
@@ -1693,25 +1720,24 @@ def static calculateBuildCommands(def newJob, def scenario, def branch, def isPR
                     // Debug runs take too long to run. So build job only.
                     if (lowerConfiguration == "debug") {
-                       buildCommands += "set __TestIntermediateDir=int&&build.cmd ${lowerConfiguration} ${architecture} toolset_dir C:\\ats2"
+                       buildCommands += "set __TestIntermediateDir=int&&build.cmd ${lowerConfiguration} ${architecture} toolset_dir C:\\ats2 -priority=1"
```
This seems unnecessary. Wont we be adding a bunch of time to the jump building tests we will never run?

BruceForstall replied:
I can revert this if you like. It just seemed weird to me that we build the pri 1 tests for checked, and for debug for arm, but not here. Why not be consistent?

jashook replied:
I think I would prefer to keep this as it is, I understand the consistency argument. However, we do not use these tests and the time added to the build is significant.

BruceForstall commented:
@dotnet-bot test CI please

BruceForstall commented:
@dotnet-bot test CI please

BruceForstall commented:
@RussKeldorph @jashook PTAL -- I've changed it so all the new stress periodic jobs for arm/armlb/arm64 are `@weekly`, not `@daily`. Also addressed PR feedback.

BruceForstall commented:
@dotnet-bot test CI please
